addons/mail/static/src/core/common/composer.js

```diff
@@ -534,9 +534,17 @@ export class Composer extends Component {
             mentionedChannels: this.props.composer.mentionedChannels,
             mentionedPartners: this.props.composer.mentionedPartners,
         });
+        let default_body = await prettifyMessageContent(body, validMentions);
+        // We need to wrap the content inside a div element before sending the content to the editor
+        // because the textarea inline content will be changed by sanitize on the onchange call
+        // and it will be wrapped inside a p because of html.fromstring in python.
+        // The wrapping inside a p will make all the content as one single block. Applying block-level
+        // commands like (headings, list..) will not work as they will be applied on the top p element
+        // which means the whole content.
+        default_body = "<div>" + default_body + "</div>";
```

Comment:
Keep comment as brief as you can, without referring to technical details that are not in immediate scope of these LOCs:

- when comments are too long, we don't want to read them, so they just pollute code and we need to scroll more to see whole context of effective code
- when comment refers to how / why with code of now, they will most likely lie in the future and this becomes even harmful. Comments are poor for maintainability, so they should be rare or very brief to give a good grasp of immediate intent here

Here's my suggestion of comment:

```js
default = "<div>" + default_body + "</div>"; // as to not wrap in <p> by html_sanitize
```

The comment is brief and gives a good grasp of the intent. It doesn't tell everything in details, the test coverage is there for more details. Most people will have enough clue of the intent and why this is relevant (or maybe not), with minimal footprint on the code.

_IMO I like inline comments, I feel they are underrated: it forces you to put comment on the actual relevant LOC and keep it as brief as possible (even can make you reconsider rewriting the code differently to not "require" any comment)._

Reply:
@alexkuhn , thanks for the review, I updated the fix now as suggested